docs/reference/TensorFlow.md

```diff
@@ -1,405 +1,370 @@
 
-# Test keyword spotting
+<span class="tips">**Tip:** Building TensorFlow and training the model will each take a couple of hours on an average computer. We will not perform this at this stage. If you are interested in the full flow, including the training of the model, refer to the [Supplementary information: model training](https://developer.arm.com/solutions/machine-learning-on-arm/developer-material/how-to-guides/build-arm-cortex-m-voice-assistant-with-google-tensorflow-lite/supplementary-information-model-training) guide.</span>
 
-The program outputs recognition results to its serial port. To see the output of the program, we will need to establish a serial connection with the board at 9600 baud.
+## Background: Converting the trained model
 
-The board’s USB UART shows up as  ```/dev/tty.usbmodemXXXXXXX```.We can use ‘screen’ to access the serial console. Although, ‘screen’ is not installed on Linux by default, you can use apt-get install screen to install the package.
+To get a converted model that can run on the controller itself from the trained model, we need to run a conversion script: the [TensorFlow Lite converter](https://www.tensorflow.org/lite/convert). This tool makes our model as small and efficient as possible, and converts it to a TensorFlow Lite FlatBuffer. To reduce the size of the model, we used a technique called [quantization](https://www.tensorflow.org/lite/performance/post_training_quantization). All weights and activations in the model get converted from 32-bit floating point format to an 8-bit and fixed-point format, as you can see in the following command:
 
-Run the following command in a separate terminal:
+<span class="images">![Convert the model to the TensorFlow Lite format code](https://raw.githubusercontent.com/COTASPAR/K66F/master/images/convert_model.png)</span>
 
-```screen /dev/tty.usbmodemXXXXXX 9600```
+This conversion will not only reduce the size of the network traffic, but will also avoid the more computationally expensive floating points.
 
-**Note**: this command may very depending on where your board is plugged.
+To save time, in this guide we skip the conversion step and instead download a converted model.
 
-Try saying the word “yes” several times. You should see some output like the following:
+## Downloading the model and creating a C file
 
-```
-Heard yes (208) @116448ms  
-Heard unknown (241) @117984ms  
-Heard no (201) @124992ms
+1. Download the [tiny_conv.tflite file](https://developer.arm.com/-/media/Files/downloads/Machine%20learning%20how-to%20guides/tiny_conv.tflite?revision=495eb362-4325-49b8-b3ba-3141df0c9b95&la=en&hash=0F37BA2C5DE95A1561979CDD18973171767A47C3).
+1. Convert this model into a C file that works with Mbed OS projects.
 
-```
+    Use a tool called xxd to generate a file called `model.cc`:
 
-Congratulations! You are now running a machine learning model that can recognize keywords on an Arm Cortex-M4 microcontroller, directly on your K66F.
+    ```
```

Comment:
What is it? What does it do?

Reply:
Already described above "Convert this model into a C file that works with Mbed OS projects.

Use a tool called xxd to generate a file called model.cc:"

Reply:
I meant, what is `xxd`? As a user, I don't want to run some arbitrary shell command on my machine without understanding what it does 😉

Reply:
Ah, I see. We could add "[xxd](https://www.tutorialspoint.com/unix_commands/xxd.htm) is a UNIX command-line utility that creates a hex dump of a given file or standard input. It can also convert a hex dump back to its original binary form."